plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetastoreAccessOperations.java

```diff
@@ -0,0 +1,198 @@
+import io.trino.plugin.hive.HdfsConfig;
+import io.trino.plugin.hive.HdfsConfiguration;
+import io.trino.plugin.hive.HdfsConfigurationInitializer;
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.HiveHdfsConfiguration;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.hive.authentication.NoHdfsAuthentication;
+import io.trino.plugin.hive.metastore.MetastoreConfig;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastoreConfig;
+import io.trino.spi.type.IntegerType;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.MaterializedResult;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+import static io.trino.plugin.iceberg.CountingAccessFileHiveMetastore.GET_ALL_DATABASES;
+import static io.trino.plugin.iceberg.CountingAccessFileHiveMetastore.GET_DATABASE;
+import static io.trino.plugin.iceberg.CountingAccessFileHiveMetastore.GET_TABLE;
+import static io.trino.plugin.iceberg.CountingAccessFileHiveMetastore.GET_TABLE_STATISTICS;
+import static io.trino.plugin.iceberg.CountingAccessFileHiveMetastore.GET_TABLE_WITH_PARAMETER;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.testing.MaterializedResult.resultBuilder;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestIcebergMetastoreAccessOperations
+        extends AbstractTestQueryFramework
+{
+    private static final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("iceberg")
+            .setSchema("test_schema")
+            .build();
+
+    private CountingAccessFileHiveMetastore metastore;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("iceberg")
+                .setSchema("test_schema")
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
+                // Tests that inspect MBean attributes need to run with just one node, otherwise
+                // the attributes may come from the bound class instance in non-coordinator node
+                .setNodeCount(1)
+                .build();
+
+        File baseDir = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data").toFile();
+        metastore = createCountingAccessesFileHiveMetaStore(baseDir);
+        queryRunner.installPlugin(new TestingIcebergPlugin(metastore, true));
+        queryRunner.createCatalog("iceberg", "iceberg");
+
+        queryRunner.execute("CREATE SCHEMA test_schema");
+        return queryRunner;
+    }
+
+    @BeforeMethod
+    public void prepare()
+    {
+        metastore.resetCounters();
```

Comment:
This is good, but requires every case to be a separate test method.
We usually don't do that to avoid many test methods (and avoid having to come up with names better than `testSelectWithPredicate1`, `testSelectWithPredicate2`, ``testSelectWithPredicate3`).

eg I would want to have ``testSelectWithPredicate` method with interesting test cases adorned with `//` comments (unless are self-explanatory)

for this, all you need is a  method like

```
assertMetastoreInvocations(String query, Multiset expectedInvocations)
```

(or 

```
assertMetastoreInvocations(Runnable callback, Multiset expectedInvocations)
```
)

Reply:
ok